### PR TITLE
Add WikiSource().__iter__() and WikiSource().__len__()

### PR DESCRIPTION
**Pull request by cclauss**
As discussed in #45, `WikiSource`, will need `iter()` and `len()` methods to be compatible with Python 3.

`web.utf8()` was removed in favor of `web.safestr()` in release v0.40
https://github.com/webpy/webpy/blob/master/ChangeLog.txt

### CODE REVIEW

File: infogami/infobase/server.py [hunk 1]
```diff
@@ -94,8 +94,8 @@ def g(self, *a, **kw):
             else:
                 process_exception(e)
 
-        
-        result = d.json_data if isinstance(d, JSON) else json.dumps(d)
+        # use default=str to deal with datetimes on Python 3
+        result = d.json_data if isinstance(d, JSON) else json.dumps(d, default=str)
```
Comment (cdrini) [inline infogami/infobase/server.py:98]: This should have errored on python2 as well; why is this necessary?

image

File: infogami/plugins/wikitemplates/code.py [hunk 2]
```diff
@@ -4,13 +4,17 @@
...
 import os
+import sys
 try:
     from collections.abc import Mapping
 except ImportError:
     from collections import Mapping
 
 import web
-
+if "pytest" in sys.modules:
```
Comment (cdrini) [inline infogami/plugins/wikitemplates/code.py:14]: Can we have a comment on this line? I'm not too sure what it's doing. If this is overriding something for testing purposes, it would be great if we could keep it in the test files, but it's ok if that's not possible :)

File: infogami/plugins/wikitemplates/code.py [hunk 3]
```diff
@@ -44,6 +48,13 @@ def __getitem__(self, key):
 
         return value
 
+    # TODO: Should __iter__() and keys() return the same keys??
+    def __iter__(self):
+        return self.templates.__iter__()
+
+    def __len__(self):
+        return len(self.templates)
+
```
Comment (cdrini) [inline infogami/plugins/wikitemplates/code.py:57]: Can you give a quick summary of these are now necessary? Is it because we switched to `Mapping`?

File: infogami/infobase/server.py (hunk 1 above)
Comment (cclauss) [reply infogami/infobase/server.py:98]: Removed these changes.  Will attack in a separate PR.

File: infogami/plugins/wikitemplates/code.py (hunk 2 above)
Comment (cclauss) [reply infogami/plugins/wikitemplates/code.py:14]: Removed these changes. Will attack in a separate PR.

File: infogami/plugins/wikitemplates/code.py (hunk 3 above)
Comment (cclauss) [reply infogami/plugins/wikitemplates/code.py:57]: http://python3porting.com/problems.html#replacing-userdict

**Comment by cclauss** [comment]
Datetime serialization issues in Py2 or Py3?

**Comment by cdrini** [comment]
Both; it seems like an issue with infogami master.
